core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowAccessResource.scala

```diff
@@ -2,221 +2,79 @@ package edu.uci.ics.texera.web.resource.dashboard.workflow
-            AccessEntry(userName, "Write")
-          } else {
-            AccessEntry(userName, "Read")
-          }
-      })
+  def hasWriteAccess(wid: UInteger, uid: UInteger): Boolean = {
+    getPrivilege(wid, uid).eq(WorkflowUserAccessPrivilege.WRITE)
   }
 
-  object WorkflowAccess extends Enumeration {
-    type Access = Value
-    val READ: WorkflowAccess.Value = Value("Read")
-    val WRITE: WorkflowAccess.Value = Value("Write")
-    val NONE: WorkflowAccess.Value = Value("None")
-    val NO_RECORD: WorkflowAccess.Value = Value("NoRecord")
+  def hasAccess(wid: UInteger, uid: UInteger): Boolean = {
+    hasReadAccess(wid, uid) || hasWriteAccess(wid, uid)
   }
 }
 
-/**
-  * Provides endpoints for operations related to Workflow Access.
-  */
 @Produces(Array(MediaType.APPLICATION_JSON))
+@RolesAllowed(Array("REGULAR", "ADMIN"))
 @Path("/workflow/access")
 class WorkflowAccessResource() {
+  final private val userDao = new UserDao(context.configuration())
+  final private val workflowOfUserDao = new WorkflowOfUserDao(context.configuration)
+  final private val workflowUserAccessDao = new WorkflowUserAccessDao(context.configuration)
 
-  private val workflowOfUserDao = new WorkflowOfUserDao(
-    context.configuration
-  )
-  private val workflowUserAccessDao = new WorkflowUserAccessDao(
-    context.configuration
-  )
-
-  /**
-    * This method returns the owner of a workflow
-    *
-    * @param wid,     the given workflow
-    * @return ownerName,  the owner's name
-    */
   @GET
   @Path("/owner/{wid}")
-  @RolesAllowed(Array("REGULAR", "ADMIN"))
-  def getWorkflowOwner(@PathParam("wid") wid: UInteger): OwnershipResponse = {
+  def getOwner(@PathParam("wid") wid: UInteger): String = {
     val uid = workflowOfUserDao.fetchByWid(wid).get(0).getUid
-    val ownerName = userDao.fetchOneByUid(uid).getName
-    OwnershipResponse(ownerName)
+    userDao.fetchOneByUid(uid).getEmail
   }
 
-  /**
-    * This method identifies the user access level of the given workflow
-    *
-    * @param wid     the given workflow
-    * @return json object indicating uid, wid and access level, ex: {"level": "Write", "uid": 1, "wid": 15}
-    */
-  @GET
-  @Path("/workflow/{wid}/level")
-  @RolesAllowed(Array("REGULAR", "ADMIN"))
-  def retrieveUserAccessLevel(
-      @PathParam("wid") wid: UInteger,
-      @Auth sessionUser: SessionUser
-  ): AccessResponse = {
-    val user = sessionUser.getUser
-    val uid = user.getUid
-    val workflowAccessLevel = checkAccessLevel(wid, uid).toString
-    AccessResponse(uid, wid, workflowAccessLevel)
-  }
-
-  /**
-    * This method returns all current shared accesses of the given workflow
-    *
-    * @param wid     the given workflow
-    * @return json object indicating user with access and access type, ex: [{"Jim": "Write"}]
-    */
   @GET
   @Path("/list/{wid}")
-  @RolesAllowed(Array("REGULAR", "ADMIN"))
-  def retrieveGrantedWorkflowAccessList(
+  def getList(
```

Comment:
You renamed the names `getList` and `AccessEntry2` in frontend but did not do so for the back end.
Also, notice how the original code had documentations for each method, and in the new code there is no such thing. It would be better to explain each function. At the very least you need to explain this `getList` here, since it has a less informative name than other methods.

Reply:
Renamed. I forgot to rename the backend. The getList method was designed to be a generic method to get all three(workflow/project/file) types of lists but it turns out that I will refactor the workflow page first.

Reply:
I don't think you have pushed your changes.

Reply:
Pushed.